portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +36,57 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  /** Returns all questions posted */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Jason</h1>");
+    Query query = new Query("Questions").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Question> questions = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String name = (String) entity.getProperty("name");
+      String question = (String) entity.getProperty("question");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Question userQuestion = new Question(name, question, timestamp);
+      questions.add(userQuestion);
+    }
+
+    response.setContentType("application/json;");
+    response.getWriter().println(convertToJSON(questions));
+  }
+
+  /** Posts the users questions on a message board. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String name = request.getParameter("author");
```

Comment:
Personally I think it's a little confusing that `name` on the frontend == `author` on the backend and `question` on the frontend == `description` on the backend. At the same time `question` is already being used for your actual `Question` objects in addition to the Question text. I would personally prefer consistent naming on the frontend and backend. 

I'll add another comment about your Question.question field though.